src/test/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessorTest.java

```diff
@@ -270,7 +274,48 @@ void GIVEN_one_component_one_file_more_than_max_WHEN_merge_THEN_reads_partial_fi
     }
 
     @Test
-    void GIVEN_one_component_one_file_24h_gap_WHEN_merge_THEN_reads_partial_file(ExtensionContext context1)
+    void GIVEN_one_component_one_file_24h_gap_WHEN_merge_THEN_reads_partial_file()
+            throws IOException {
+        File file = new File(directoryPath.resolve("greengrass_test.log").toUri());
+        assertTrue(file.createNewFile());
+        assertTrue(file.setReadable(true));
+        assertTrue(file.setWritable(true));
+        Instant now = Instant.now();
+        Instant then = now.minus(1, ChronoUnit.DAYS);
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH);
+        try (OutputStream fileOutputStream = Files.newOutputStream(file.toPath())) {
+            fileOutputStream.write((sdf.format(new Date(then.toEpochMilli())) + "T01:00:00Z ABC1\n").getBytes(StandardCharsets.UTF_8));
+            fileOutputStream.write((sdf.format(new Date(then.toEpochMilli())) + "T02:00:00Z ABC2\n").getBytes(StandardCharsets.UTF_8));
+            fileOutputStream.write((sdf.format(new Date(now.toEpochMilli())) + "T01:00:00Z ABC3\n").getBytes(StandardCharsets.UTF_8));
+            fileOutputStream.write((sdf.format(new Date(now.toEpochMilli())) + "T02:00:00Z ABC4\n").getBytes(StandardCharsets.UTF_8));
+        }
+
+        try {
+            List<LogFileInformation> logFileInformationSet = new ArrayList<>();
+            logFileInformationSet.add(LogFileInformation.builder().startPosition(0).file(file).build());
+            ComponentLogFileInformation componentLogFileInformation = ComponentLogFileInformation.builder()
+                    .name("TestComponent")
+                    .multiLineStartPattern(Pattern.compile("^[^\\s]+(\\s+[^\\s]+)*$"))
+                    .desiredLogLevel(Level.INFO)
+                    .componentType(ComponentType.GreengrassSystemComponent)
+                    .logFileInformationList(logFileInformationSet)
+                    .build();
+
+            logsProcessor = new CloudWatchAttemptLogsProcessor(mockDeviceConfiguration);
+            CloudWatchAttempt attempt = logsProcessor.processLogFiles(componentLogFileInformation);
+            assertNotNull(attempt);
+            sdf = new SimpleDateFormat("/yyyy/MM/dd/'thing/testThing'", Locale.ENGLISH);
+            String logStream1 = sdf.format(new Date(then.toEpochMilli()));
+            String logStream2 = sdf.format(new Date(now.toEpochMilli()));
+            assertThat(attempt.getLogStreamsToLogEventsMap().get(logStream1).getLogEvents(), hasSize(2));
+            assertThat(attempt.getLogStreamsToLogEventsMap().get(logStream2).getLogEvents(), hasSize(2));
+        } finally {
+            assertTrue(file.delete());
+        }
+    }
+
+    @Test
+    void GIVEN_one_component_one_file_older_than_14_days_WHEN_skip_file()
```

Comment:
nit: `GIVEN_one_component_one_file_older_than_14_days_WHEN_skip_file` _THEN ??

Or is this supposed to be: `GIVEN_one_component_WHEN_file_older_than_14_days_THEN_skip_file`

Reply:
This also doesn't seem to fully test the change - you have it filter out log events that are > 14 days.

Is it possible to have a log file with events that are > 14 days old and some that are < 14 days old?